### PR TITLE
How to solve initial value problem using ode45 in Matlab

### LINKED ISSUE #3170: How to solve initial value problem using ODE45 in Matlab
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Introduction:
MATLAB’s standard solver for ordinary differential equations (ODEs) is the function ode45. This function implements a Runge-Kutta method with a variable time step for efficient computation. ode45 is designed to handle the following general problem:
dx/dt = f(t,x),                           x(t0)  = xo
where t is the independent variable, x is a vector of dependent variables to be found and f (t, x) is a function of t and x. The mathematical problem is specified when the vector of functions on the right-hand side of Eq. (1), f (t, x), is set and the initial conditions,
x = x 0 at time t 0 are given. 
### Key takeaways: 
- Basic understanding of the ODE45
- Implementation of ODE45 in Matlab
- Applications of ODE45
### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by josephodhis**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3170

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by hectorkambow** [comment]
@josephodhis 

[How to solve initial value problem using ode45 in Matlab #3689.pdf](https://github.com/section-engineering-education/engineering-education/files/7118025/How.to.solve.initial.value.problem.using.ode45.in.Matlab.3689.pdf)


👋 Good afternoon - I'll be running a final review on your PR when ready.
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

File: content/articles/how-to-solve-initial-value-problem-using-ode45-in-matlab/index.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  -url: /engineering-education/how-to-solve-initial-value-problem-using-ode45-in-matlab/hero.jpg
+   alt: MATLAB solve Initial Value Problem example image
+---
+
+
+### Introduction
+
+MATLAB’s uses the `ode45` function as the standard solver for ordinary differential equations. The `ode45` function applies Runga-Kutta formulae with the time step variable for easy computation. `ode45` is meant to solve the equation of the general form shown below;
+
+dx/dt = f(t,x), x(t0) = xo
```
Comment (lalith1403) [inline content/articles/how-to-solve-initial-value-problem-using-ode45-in-matlab/index.md:23]: Use Latex to display mathematical equations.

File: content/articles/how-to-solve-initial-value-problem-using-ode45-in-matlab/index.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+Here, `t` is the independent variable, `x` is a vector of dependent variables to be found, and f (t, x) is a function of t and x. The mathematical equation is defined when the `x` of functions on the right-hand side of Eq. (1), f (t, x), is pre-determined, and the initial conditions, x = x0 at time t0, are given.
```
Comment (lalith1403) [inline content/articles/how-to-solve-initial-value-problem-using-ode45-in-matlab/index.md:25]: Use Inline latex to denote mathematical terms. Refer to the earlier published article on MATLAB.

File: content/articles/how-to-solve-initial-value-problem-using-ode45-in-matlab/index.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+In this article, we will consider the Matlab algorithm `ode45` to solve the IVP. IVP is an ODE together with some initial value(s). Here you have a differential equation describing the dynamics of how something is changing, but you know where to start. The initial value tells you where to start, but the differential equation tells you where to go next. `ode45` is a work course in Matlab, and it's the first algorithm you go to solve ode problems.
```
Comment (lalith1403) [inline content/articles/how-to-solve-initial-value-problem-using-ode45-in-matlab/index.md:27]: First occurrence of abbreviation should contain the full form as well.

File: content/articles/how-to-solve-initial-value-problem-using-ode45-in-matlab/index.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+To follow along with this tutorial, you'll need:
+- [MATLAB](https://www.mathworks.com/products/get-matlab.html?s_tid=gn_getml) installed.
+- Proper understanding of [MATLAB](https://www.section.io/engineering-education/getting-started-with-matlab/) basics.
+
+In Runga Kutta 4 method, we obtain the solution of $y_i$ at various values of $t_i$ between the range $t_0$ and $t_n$. The intermediate value at which the computations are done is calculated internally by the ODE45. ODE45 uses adaptive step size Runga Kutta 4 method. This indicates that stepsize `H` is a varying value. It means that you have to define the stepsize.  Step size is the distance between the current value and the subsequent initial value.
```
Comment (lalith1403) [inline content/articles/how-to-solve-initial-value-problem-using-ode45-in-matlab/index.md:33]: Explain Runge Kutta method. Check the spelling. It is Runge Kutta.

File: content/articles/how-to-solve-initial-value-problem-using-ode45-in-matlab/index.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+![error defination](ode_two.png)
+
+The error indicates that the `t1` is undefined since the input variable was `t`. Keep in mind that our `t` and `y` are dummy variables. It means that they don't have any specification name, but you can use any name. The only thing that you should note is to use a similar name for the input and, in the definition, your function `dy`. Then, it won't be a problem. Also, as you use a different name, the order of the variable must remain, independent variable followed by the independent.
+
+### Plug flow reactor(PFR)
+
+Plug flow reactor, also known as tabular reactor, consists of a cylindrical pipe with an opening on each end for the reactor and products to flow through. Consumption of reactors is continuous as they flow down the length of the reactors. It means that as the reaction continues, the products increases.
+
+The plug flow reactor may be figured as one long tube or several short tubes. They range in diameter from few centimetres to several meters. The choice of diameter is based on the construction cost, pumping cost, and heat transfer needed. They have a wide variety of applications in either gas or liquid phase systems.
```
Comment (lalith1403) [inline content/articles/how-to-solve-initial-value-problem-using-ode45-in-matlab/index.md:106]: PFR needs through explanation. This explanation does not guide the reader on what the problem is.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
@josephodhis Welcome to Eng-Ed program. This is a wonderful article with thought provoking examples. Can you add more explanation wherever you are explaining the problem and the code. Once that is done, we can move to the final review. 

Also, perform a periodic check for grammatical errors using services such as Grammarly.

**Comment by lalith1403** [comment]
@josephodhis Any updates on this?

**Comment by josephodhis** [comment]
@lalith1403 changes made.

**Comment by LinusMuema** [comment]
> @josephodhis
> 
> [How to solve initial value problem using ode45 in Matlab #3689.pdf](https://github.com/section-engineering-education/engineering-education/files/7118025/How.to.solve.initial.value.problem.using.ode45.in.Matlab.3689.pdf)
> 
>  Good afternoon - I'll be running a final review on your PR when ready.
> Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
> Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
> Be sure to see our resources page to see more info on plagiarism and what is considered as such.

Hi @hectorkambow could you check the plagiarism issues you raised here to check if the article is ready for a final review...

**Comment by josephodhis** [comment]
@LinusMuema the plagarism issue is corrected. please check

**Comment by josephodhis** [comment]
@hectorkambow would you please check this.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@josephodhis 
Please be sure to go through the edits made by your peer reviewer and content moderator to improve future articles.

As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published.
Our team of peer reviewers do not provide revisions services at this time in order to review them all in a timely manner. 

All revisions and edits should be completed on your own forked repo (as to not take up room in the queue). 
Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

cc @lalith1403 
Please go through the edits as well. 👍

**Pull request merged**